lib/std/zon/parse.zig

```diff
@@ -0,0 +1,2832 @@
+            .nan => return std.math.nan(T),
+            .char_literal => |val| return @floatFromInt(val),
+            else => return self.failNodeFmt(node, "expected type '{s}'", .{@typeName(T)}),
+        }
+    }
+
+    fn parseEnumLiteral(
+        self: @This(),
+        comptime T: type,
+        node: Zoir.Node.Index,
+    ) !T {
+        switch (node.get(self.zoir)) {
+            .enum_literal => |string| {
+                // Create a comptime string map for the enum fields
+                const enum_fields = @typeInfo(T).@"enum".fields;
+                comptime var kvs_list: [enum_fields.len]struct { []const u8, T } = undefined;
+                inline for (enum_fields, 0..) |field, i| {
+                    kvs_list[i] = .{ field.name, @enumFromInt(field.value) };
+                }
+                const enum_tags = std.StaticStringMap(T).initComptime(kvs_list);
+
+                // Get the tag if it exists
+                return enum_tags.get(string.get(self.zoir)) orelse
+                    self.failUnexpectedField(T, node, null);
+            },
+            else => return self.failNode(node, "expected enum literal"),
+        }
+    }
+
+    fn parsePointer(
+        self: *@This(),
+        comptime T: type,
+        comptime options: Options,
+        node: Zoir.Node.Index,
+    ) !T {
+        switch (node.get(self.zoir)) {
+            .string_literal => return try self.parseString(T, node),
+            .array_literal => |nodes| return try self.parseSlice(T, options, nodes),
+            .empty_literal => return try self.parseSlice(T, options, .{ .start = node, .len = 0 }),
+            else => return self.failExpectedContainer(T, node),
+        }
+    }
+
+    fn parseString(
+        self: *@This(),
+        comptime T: type,
+        node: Zoir.Node.Index,
+    ) !T {
+        const ast_node = node.getAstNode(self.zoir);
+        const pointer = @typeInfo(T).pointer;
+        var size_hint = ZonGen.strLitSizeHint(self.ast, ast_node);
+        if (pointer.sentinel_ptr != null) size_hint += 1;
+
+        var buf: std.ArrayListUnmanaged(u8) = try .initCapacity(self.gpa, size_hint);
+        defer buf.deinit(self.gpa);
+        switch (try ZonGen.parseStrLit(self.ast, ast_node, buf.writer(self.gpa))) {
+            .success => {},
+            .failure => |err| {
+                const token = self.ast.nodes.items(.main_token)[ast_node];
+                const raw_string = self.ast.tokenSlice(token);
+                return self.failTokenFmt(token, @intCast(err.offset()), "{s}", .{err.fmt(raw_string)});
+            },
+        }
+
+        if (pointer.child != u8 or
+            pointer.size != .slice or
+            !pointer.is_const or
+            (pointer.sentinel_ptr != null and @as(*const u8, @ptrCast(pointer.sentinel_ptr)).* != 0) or
+            pointer.alignment != 1)
+        {
+            return self.failExpectedContainer(T, node);
+        }
+
+        if (pointer.sentinel_ptr != null) {
```

Comment:
```suggestion
        if (pointer.sentinel() != null) {
```
Exactly equivalent, I just personally feel that this is a little clearer.